tests/nodes/tests_Node.cpp

```diff
@@ -360,11 +385,36 @@ TEST_CASE("variable inlining") {
         CHECK_EQ(var.backend_handle().id(), again3.backend_handle().id());
     };
 
+    auto const check_fetch_or_serialize = [&](query::Variable const &var) {
```

Comment:
`check_fetch_or_serialize` should have another parameter `FetchOrSerializeResult expected_result` to validate the outcome. Also applies to `check_fetch_or_serialize` for `BlankNode`s.